lisa/sut_orchestrator/azure/features.py

```diff
@@ -783,3 +787,117 @@ def get_azure_disk_type(disk_type: schema.DiskType) -> str:
     schema.DiskType.StandardHDDLRS: "Standard_LRS",
     schema.DiskType.StandardSSDLRS: "StandardSSD_LRS",
 }
+
+
+class Resize(AzureFeatureMixin, features.Resize):
+    def resize(
+        self, resize_action: ResizeAction = ResizeAction.IncreaseCoreCount
+    ) -> schema.NodeSpace:
+        platform: AzurePlatform = self._platform  # type: ignore
+        compute_client = get_compute_client(platform)
+        node_context = get_node_context(self._node)
+        new_vm_size_info = self._select_vm_size(resize_action)
+
+        # Creating parameter for VM Operations API call
+        hardware_profile = HardwareProfile(vm_size=new_vm_size_info.vm_size)
+        vm_update = VirtualMachineUpdate(hardware_profile=hardware_profile)
+
+        # Resizing with new Vm Size
+        compute_client.virtual_machines.begin_update(
```

Comment:
I see it's a "begin" resize. Make sure the resize will be done before exit this method. Use the wait_operation method to wait it done.

Reply:
done